modules/openapi-generator/src/main/resources/Groovy/api.mustache

```diff
@@ -49,38 +49,38 @@ class {{classname}} {
         contentType = '{{{mediaType}}}';
         {{/consumes.0}}
         {{/bodyParam}}
-        {{#bodyParams.0}}
-        {{^hasMore}}
-        bodyParams = {{paramName}}
-        {{/hasMore}}
-        {{#hasMore}}
+        {{#bodyParams}}
+        {{#-first}}
+        {{^-last}}
         bodyParams = [:]
         bodyParams.put("{{baseName}}", {{paramName}})
-        {{/hasMore}}
-        {{/bodyParams.0}}
-        {{#bodyParams}}
-        {{#secondaryParam}}
+        {{/-last}}
+        {{#-last}}
+        bodyParams = {{paramName}}
+        {{/-last}}
+        {{/-first}}
+        {{^-first}}
         bodyParams.put("{{baseName}}", {{paramName}})
-        {{/secondaryParam}}
+        {{/-first}}
```

Comment:
This is so fancy the end result is:
if we have only one item in bodyParams use:
`bodyParams = ParamA`
If we have multiple bodyParams we get:
```
bodyParams. = [:]  # empty map in Groovy
bodyParams.put("ParamA", ParamA
bodyParams.put("ParamB", ParamB
```

Reply:
I think other client generators do not support multiple body parameters so as long as the single body parameter use case works fine, I think that should be good enough.

Reply:
Good point. I updated the template to only use bodyParam.